### PR TITLE
fall-forward status + auto start fall-forward synchronize

### PR DESCRIPTION
**Pull request by makalaaneesh**

### CODE REVIEW

File: yb-voyager/cmd/importData.go [hunk 1]
```diff
@@ -94,6 +96,68 @@ func importDataCommandFn(cmd *cobra.Command, args []string) {
...
+	fallForwardSynchronizeCmdStr := strings.Join(fallForwardSynchronizeCmd, " ")
+	if utils.DoNotPrompt {
+		fallForwardSynchronizeCmd = append(fallForwardSynchronizeCmd, "--yes")
+	}
+	if disablePb {
+		fallForwardSynchronizeCmd = append(fallForwardSynchronizeCmd, "--disable-pb")
+	}
+
+	utils.PrintAndLog("Starting ff synchronize with command %s", fallForwardSynchronizeCmdStr)
+	binary, lookErr := exec.LookPath("yb-voyager")
```
[the remark this answers is not in the document]
Comment (makalaaneesh) [reply yb-voyager/cmd/importData.go]: exec.LookPath works fine with os.Args[0] 👍 It basically gets the abs path.

```
~/Documents/yb-voyager/yb-voyager (aneesh/ff-cmds*) » ~/go/bin/yb-voyager version                                                                                      amakala@am-mbp-2qfp0
INFO[0000] args[0] = /Users/amakala/go/bin/yb-voyager|executablePath =/Users/amakala/go/bin/yb-voyager

~/Documents/yb-voyager/yb-voyager (aneesh/ff-cmds*) » yb-voyager version                                                                                               amakala@am-mbp-2qfp0
INFO[0000] args[0] = yb-voyager|executablePath =/usr/local/bin/yb-voyager 
```

File: yb-voyager/cmd/importData.go [hunk 2]
```diff
@@ -94,6 +96,68 @@ func importDataCommandFn(cmd *cobra.Command, args []string) {
...
+	if !migrationStatusRecord.FallForwarDBExists {
+		utils.PrintAndLog("no ff db exists. returning. ")
+		return
+	}
+	// TODO: ssl* params
+	// TODO: obfuscate password.
+	fallForwardSynchronizeCmd := []string{"yb-voyager", "fall-forward", "synchronize",
+		"--export-dir", exportDir,
+		"--source-db-host", tconf.Host,
+		"--source-db-port", fmt.Sprintf("%d", tconf.Port),
```
[the remark this answers is not in the document]
Comment (makalaaneesh) [reply yb-voyager/cmd/importData.go:123]: Yes, yes, it's in my TODO list. Will raise a separate PR

File: yb-voyager/cmd/importData.go [hunk 3]
```diff
@@ -122,42 +122,31 @@ func startFallforwardSynchronizeIfRequired(tableList []string) {
...
 	}
 	if disablePb {
 		fallForwardSynchronizeCmd = append(fallForwardSynchronizeCmd, "--disable-pb")
 	}
+	fallForwardSynchronizeCmdStr := "SOURCE_DB_PASSWORD=*** " + strings.Join(fallForwardSynchronizeCmd, " ")
 
-	utils.PrintAndLog("Starting ff synchronize with command %s", fallForwardSynchronizeCmdStr)
-	binary, lookErr := exec.LookPath("yb-voyager")
+	utils.PrintAndLog("Starting fall-forward synchronize with command:\n %s", color.GreenString(fallForwardSynchronizeCmdStr))
+	binary, lookErr := exec.LookPath(os.Args[0])
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:140]: Should we be using `AbsPath()`?

File: yb-voyager/cmd/fallForwardStatus.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	},
+}
+
+func init() {
+	fallForwardCmd.AddCommand(fallforwardStatusCmd)
+	fallforwardStatusCmd.Flags().StringVarP(&exportDir, "export-dir", "e", "",
+		"export directory is the workspace used to keep the exported schema, data, state, and logs")
+}
+
+func checkAndReportFallforwardStatus() {
```
[the remark this answers is not in the document]
Comment (amit-yb) [reply yb-voyager/cmd/fallForwardStatus.go:43]: Missed this change?

File: yb-voyager/cmd/importData.go [hunk 5]
```diff
@@ -94,6 +96,57 @@ func importDataCommandFn(cmd *cobra.Command, args []string) {
 	importFileTasks := discoverFilesToImport()
 	importFileTasks = applyTableListFilter(importFileTasks)
 	importData(importFileTasks)
+	var tablelist []string
+	for _, ift := range importFileTasks {
+		tablelist = append(tablelist, ift.TableName)
+	}
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:102]: There is already a function to do this task: `importFileTasksToTableNames()`

File: yb-voyager/cmd/importData.go [hunk 6]
```diff
@@ -94,6 +96,57 @@ func importDataCommandFn(cmd *cobra.Command, args []string) {
...
+	startFallforwardSynchronizeIfRequired(tablelist)
+}
+
+func startFallforwardSynchronizeIfRequired(tableList []string) {
+	if importerRole != TARGET_DB_IMPORTER_ROLE {
+		return
+	}
+	migrationStatusRecord, err := GetMigrationStatusRecord()
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:110]: Consider using shorter names for local variables in small functions. e.g. `msr`

File: yb-voyager/cmd/importData.go [hunk 7]
```diff
@@ -94,6 +96,57 @@ func importDataCommandFn(cmd *cobra.Command, args []string) {
...
+	if err != nil {
+		utils.ErrExit("could not fetch MigrationstatusRecord: %w", err)
+	}
+	if !migrationStatusRecord.FallForwarDBExists {
+		utils.PrintAndLog("No fall-forward db exists. Exiting.")
+		return
+	}
+	// TODO: ssl* params
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:118]: We should complete this before the release.

File: yb-voyager/cmd/importData.go [hunk 8]
```diff
@@ -94,6 +96,57 @@ func importDataCommandFn(cmd *cobra.Command, args []string) {
...
+	// TODO: obfuscate password.
+	fallForwardSynchronizeCmd := []string{"yb-voyager", "fall-forward", "synchronize",
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:120]: How about using `cmd` as variable name? `fall-forward synchronize` is the only command we are dealing in this function.
Sometimes using shorter/crisper variable names can in-fact improve the readability of program.

File: yb-voyager/cmd/importData.go [hunk 9]
```diff
@@ -94,6 +96,57 @@ func importDataCommandFn(cmd *cobra.Command, args []string) {
...
+		fmt.Sprintf("--send-diagnostics=%t", callhome.SendDiagnostics),
+	}
+
+	if utils.DoNotPrompt {
+		fallForwardSynchronizeCmd = append(fallForwardSynchronizeCmd, "--yes")
+	}
+	if disablePb {
+		fallForwardSynchronizeCmd = append(fallForwardSynchronizeCmd, "--disable-pb")
+	}
+	fallForwardSynchronizeCmdStr := "SOURCE_DB_PASSWORD=*** " + strings.Join(fallForwardSynchronizeCmd, " ")
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:137]: `strCmd`

File: yb-voyager/cmd/importData.go [hunk 10]
```diff
@@ -94,6 +96,57 @@ func importDataCommandFn(cmd *cobra.Command, args []string) {
...
+	utils.PrintAndLog("Starting fall-forward synchronize with command:\n %s", color.GreenString(fallForwardSynchronizeCmdStr))
+	binary, lookErr := exec.LookPath(os.Args[0])
+	if lookErr != nil {
+		utils.ErrExit("could not find yb-voyager - %w", err)
+	}
+	env := os.Environ()
+	env = append(env, fmt.Sprintf("SOURCE_DB_PASSWORD=%s", tconf.Password))
+	execErr := syscall.Exec(binary, fallForwardSynchronizeCmd, env)
+	if execErr != nil {
+		utils.ErrExit("could not successfully run yb-voyager fall-forward synchronize - %w\n Please re-run with command :\n%s", err, fallForwardSynchronizeCmdStr)
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:148]: `failed to run ...`

**Review by amit-yb — APPROVED** [review]
LGTM with a few minor comments.

File: yb-voyager/cmd/importData.go (hunk 3 above)
Comment (makalaaneesh) [reply yb-voyager/cmd/importData.go:139]: LookPath returns the abs path. 
check https://github.com/yugabyte/yb-voyager/pull/1036#discussion_r1314695347

File: yb-voyager/cmd/fallForwardStatus.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (makalaaneesh) [reply yb-voyager/cmd/fallForwardStatus.go]: Was a small function to begin with so I thought we could defer this :) 
Fixed it now anyway!

**Pull request merged**